atlp-rwanda/dynamites-ecomm-fe
# creating passwrod reset functionality

**Summary**
This pull request introduces significant updates to the authentication flow within our application, specifically focusing on the password recovery and reset functionalities. By refining these processes, we aim to provide a more secure and user-friendly experience for our users when they need to recover access to their accounts

**Changes**

- **Password Recovery Initiation:** Implemented an asynchronous thunk action, `requestPasswordReset`, which sends a request to the backend with the user's email. This initiates the password recovery process by generating a recovery token and sending it to the user's email address (this needs a fix).

- **Password Reset:** Added another asynchronous thunk action, `resetPassword`, which allows users to reset their password using the recovery token sent to their email. This action sends the token and the new password to the backend, where the token is verified, and the password is updated.

**Technical Details**

- Both thunk actions utilize [axios](https://github.com/axios/axios) for making HTTP POST requests to the backend.

- Error handling is implemented to return the error response data using `rejectWithValue` in case of request failures, ensuring that the application can gracefully inform the user about any issues encountered during the process.

**Testing**

1. Checkout this branch by running `git switch 59-implementation-for-forgot-password-functionality` in bash terminal (code is expected to work also on Windows)
2. Run `npm run dev` comand in your terminal
3. go to `[/forgot-password](http://localhost:5173/forgot-password)` in your browser (expected to work accross browsers)
4. Enter your email for the account you want to reset password for
5. click on the link sent to your inbox for password reset. It should redirect you to a page like this `http://localhost:5173/reset-password/{token}` where token is a unique key to help you reset your password
6. Fill the form with your new password two times and click submit
7. You will be redirected to login page if reset password is successful or you will see an error message.

**Screenshots**
![image](https://github.com/atlp-rwanda/dynamites-ecomm-fe/assets/85235653/506d07a3-51b0-4357-a4e6-6ed3266e3061)
![image](https://github.com/atlp-rwanda/dynamites-ecomm-fe/assets/85235653/d95b95eb-bd43-424f-a722-77153bd7dac4)
![image](https://github.com/atlp-rwanda/dynamites-ecomm-fe/assets/85235653/cd5ed4b3-694e-4ee4-bf8d-a28745b1e718)
![image](https://github.com/atlp-rwanda/dynamites-ecomm-fe/assets/85235653/c13894d8-592e-43e5-8c4a-c0de43d81028)
